src/lib/Autodiff.hs

```diff
@@ -259,6 +262,14 @@ linearizeHof env hof = case hof of
   RunWriter     lam -> linearizeEff Nothing    lam True  (const RunWriter) (emitRunWriter "r") Writer
   RunReader val lam -> linearizeEff (Just val) lam False RunReader         (emitRunReader "r") Reader
   RunState  val lam -> linearizeEff (Just val) lam True  RunState          (emitRunState  "r") State
+  RunIO ~(Lam (Abs _ (arrow, body))) -> LinA $ do
+    arrow' <- substEmbed env arrow
+    -- TODO: consider the possibility of other effects here besides IO
+    lam <- buildLam (Ignore UnitTy) arrow' $ \_ ->
+      tangentFunAsLambda $ linearizeBlock env body
+    result <- emit $ Hof $ RunIO lam
+    (ans, linLam) <- fromPair result
+    return (ans, applyLinToTangents linLam)
```

Comment:
Again, do we really need that? It's not like we can usefully differentiate anything that uses IO, and I would hope that our activity analysis is sufficient to realize that AD doesn't need to traverse into those blocks.

Reply:
The problem is that we generate unsafeIO-wrapped `load` calls in `Imp` for tables that are views of existing data. Once we have custom derivatives again we can wrap it up with a custom (zero) derivative and we shouldn't need this.

Reply:
I still don't understand why would it be necessary. Those constant tables should just get detected as inactive values, and shouldn't be linearized.

Reply:
Agree. The fact this is necessary implies a bug in activity analysis. Filed as #430 .